### ISSUE TITLE
Bounds for Draggable inside another Draggable

### ISSUE THREAD

**Issue by wahlstedt** [react-grid-layout/react-draggable#552]
The `bounds` of a Draggable inside another Draggable will be offset when the outer Draggable is moved. Is this the expected behaviour and if so, how can it be avoided?

Shouldn't the inner Draggable always be limited to the specified node, regardless of the outer Draggables current position?

Simple sandbox demonstration:
https://codesandbox.io/s/react-draggable-bounds-htnuy

(The inner draggable ("Drag Me") will have its movement limited correctly as long as the outer draggable (pink square) isn't moved, but once it is moved the limits will be offset.

**Comment by STRML** [comment]
Yes, because the inner draggable is placed relative to the outer draggable. if you don't want this, then place them side by side and use the lexical ordering in the html or a z-index to determine visual ordering.

**Comment by wahlstedt** [comment]
Thanks for the quick reply! My use case is multiple tool palettes that I want moved around as a group, which is why they are located inside another Draggable.

I get that the inner draggable is _placed_ relative to the outer one (which is what I want for my use case!) But I don't think it's logical that the `bounds` should become relative as well when they are defined as a selector. If the `bounds` were set as an {left, top, right, bottom} object I could understand the logic of the bounds moving as well. The documentation also makes it sound as if this is the way it works: _"a selector, restricts movement within the targeted node"_.

I guess I'm making a feature request to allow for `bounds` to not be relative when set as a selector. :)

**Comment by wahlstedt** [comment]
I got an headache from those bounds calculations. lol But it only seems to use offsets for the calculations... Wouldn't there be a need to use the location relative to the viewport? 

Maybe it's quicker if we have (or convert to) the absolute x,y coordinates and check those against the `getBoundingClientRect()` of the element?

**Comment by sankarnarayanansr** [comment]
You can restrict the draggable bound dynamically using the elements width and height. Get offset height and width using a ref
